File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        Queue <int[]> q = new LinkedList<>();
        q.add(new int[] {numbers[0], 0});
        q.add(new int[] {-numbers[0], 0});
        
        int N = numbers.length;
        while (!q.isEmpty())
        {
            int [] temp = q.poll();
            if (temp[1] == N - 1)
            {
                if (temp[0] == target) answer++;
                continue;
            }
            temp[1]++;
            q.add(new int[] {temp[0] + numbers[temp[1]], temp[1]});
            q.add(new int[] {temp[0] - numbers[temp[1]], temp[1]});
        }
        return answer;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    def dfs(n,o):
        global answer
        if n == len(numbers):
            if (o == target):
                answer += 1
            return
        
        dfs(n+1, o + numbers[n])
        dfs(n+1, o - numbers[n])
        return
    
    dfs(0,0)
    return answer